File: data/src/main/java/com/microsoft/azure/kusto/data/exceptions/KustoClientInvalidConnectionStringException.java
package com.microsoft.azure.kusto.data.exceptions;

/**
 * Raised when Kusto client is initialized with an invalid endpoint
 */
public class KustoClientInvalidConnectionStringException extends DataClientException {
    public KustoClientInvalidConnectionStringException(Exception e) {
        super(e);
    }

    public KustoClientInvalidConnectionStringException(String msg) {
        super(msg);
    }

    public KustoClientInvalidConnectionStringException(String clusterURL, String msg, Exception e) {
        super(clusterURL, msg, e);
    }
}

File: ingest-v2/src/main/kotlin/com/microsoft/azure/kusto/ingest/v2/exceptions/KustoClientInvalidConnectionStringException.kt
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.
package com.microsoft.azure.kusto.ingest.v2.exceptions

/**
 * Exception thrown when a connection string is invalid or the endpoint is not
 * trusted.
 */
class KustoClientInvalidConnectionStringException : RuntimeException {
    constructor(message: String) : super(message)

    constructor(
        uri: String,
        message: String,
        cause: Throwable,
    ) : super("Invalid connection string for URI '$uri': $message", cause)
}
